uaclient/files/notices.py

```diff
@@ -0,0 +1,206 @@
+import os
+from collections import namedtuple
+from enum import Enum
+from typing import Any, Dict, List, Optional  # noqa: F401
+
+from uaclient import defaults, event_logger, exceptions, system
+from uaclient.files import UAFile
+
+event = event_logger.get_event_logger()
+NoticeFileDetails = namedtuple(
+    "NoticeFileDetails", ["order_id", "label", "is_permanent"]
+)
+
+
+class Singleton(type):
+    _instances = {}  # type: Dict[Any, Any]
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(
+                *args, **kwargs
+            )
+        return cls._instances[cls]
+
+
+class NoticesDetails(NoticeFileDetails, Enum):
+    REBOOT_REQUIRED = NoticeFileDetails(
+        label="reboot_required", order_id="10", is_permanent=False
+    )
```

Comment:
Can we re-investigate including the notice message itself here? I think `.format` doesn't modify the value, it just returns a new string that is formatted. So I think it'd be okay to include in a Tuple,

Then we could format notices like:
```python
def add(notice, **kwargs):
    msg = notice.value.msg.format(**kwargs)

notices.add(NoticeDetails.REBOOT_REQUIRED, format_arg=format_val)
```

Reply:
+1 I think messages should definitely be in the Enum, and the `add` method can be responsible for the formatting. For messages where the field depends on the caller, Grant's suggestion should work fine.